vulnerabilities/importers/github.py

```diff
@@ -0,0 +1,342 @@
+                                resolved_package_urls=unaffected_purls,
+                                reference_ids=ref_ids,
+                            )
+                        )
+        return adv_list
+
+    @staticmethod
+    def categorize_versions(
+        version_range: str, all_versions: Set[str]
+    ) -> Tuple[Set[str], Set[str]]:
+        version_range = RangeSpecifier(version_range)
+        affected_versions = {
+            version for version in all_versions if version in version_range
+        }
+        return (affected_versions, all_versions - affected_versions)
+
+
+class MavenVersionAPI:
+    def __init__(self):
+        self.cache = {}
+
+    def get(self, pkg_name: str) -> Set[str]:
+        return self.cache.get(pkg_name, set())
+
+    def load_to_api(self, pkg_name: str) -> None:
+
+        if pkg_name in self.cache:
+            return
+
+        artifact_comps = pkg_name.split(':')
+        endpoint = self.artifact_url(artifact_comps)
+        resp = requests.get(endpoint).content
+
+        try:
+
+            xml_resp = ET.ElementTree(ET.fromstring(resp.decode('utf-8')))
+            self.cache[pkg_name] = self.extract_versions(xml_resp)
+
+        except ET.ParseError:
+            self.cache[pkg_name] = set()
+
+    @staticmethod
+    def artifact_url(artifact_comps: List[str]) -> str:
+
+        base_url = 'https://repo.maven.apache.org/maven2/{}'
+        group_id, artifact_id = artifact_comps
+        group_url = group_id.replace('.', '/')
+        suffix = group_url + '/' + artifact_id + '/' + 'maven-metadata.xml'
+        endpoint = base_url.format(suffix)
+
+        return endpoint
+
+    @staticmethod
+    def extract_versions(xml_response: ET.ElementTree) -> Set[str]:
+
+        all_versions = set()
+        for child in xml_response.getroot().iter():
+            if child.tag == 'version':
+                all_versions.add(child.text)
+
+        return all_versions
+
+
+class NugetVersionAPI:
+    def __init__(self):
+        self.cache = {}
+
+    def get(self, pkg_name: str) -> Set[str]:
+        return self.cache.get(pkg_name.lower(), set())
+
+    def load_to_api(self, pkg_name: str) -> None:
+        if pkg_name in self.cache:
+            return
+        endpoint = self.nuget_url(pkg_name)
+        try:
+            resp = requests.get(endpoint).json()
+        # pkg_name=Microsoft.NETCore.UniversalWindowsPlatform triggers
+        # JSONDecodeError.
+        except json.decoder.JSONDecodeError:
+            self.cache[pkg_name.lower()] = set()
+            return
+
+        self.cache[pkg_name.lower()] = self.extract_versions(resp)
+
+    @staticmethod
+    def nuget_url(pkg_name: str) -> str:
+        base_url = 'https://api.nuget.org/v3/registration5-semver1/{}/index.json'
+        return base_url.format(pkg_name.lower())
+
+    @staticmethod
+    def extract_versions(json_resp: dict) -> Set[str]:
+        all_versions = set()
```

Comment:
Why a set vs. a list? do you expect duplicates? and would using a set means that any original order may be lost?

Reply:
>do you expect duplicates?

No

> and would using a set means that any original order may be lost?

Yes, I use sets wherever  order doesn't matter, the rationale behind being I'd rather have guaranteed  unique elements at the cost of  losing the order which I don't care about here. 

If there are any significant performance gains(I will research about this) , I will switch to lists instead.

Reply:
The performance does not matter here, this is more about ordered and unique vs. unordered
I think that the order may matter though as this is the version orders?

Reply:
Order doesn't matter here.